components/sup/protocols/ctl.proto

```diff
@@ -0,0 +1,94 @@
+syntax = "proto2";
+
+import "types.proto";
+
+// Networked progress bar for displaying a remote request's operation status over time.
+message NetProgress {
+  // Number of total units until bar is complete.
+  optional uint64 total = 1;
```

Comment:
Since we're doing a binary protocol, I assume that means we care about efficiency. Isn't a `u64` overkill here? What sort of values to we expect `NetProgress` to take on? Even if we were measuring something to display on a 5K monitor, a 32-bit value would still give us over 800,000 values for each pixel.

Or maybe I'm totally missing something, like we want to be able to communicate the progress of a download in absolute bytes and that thing could be terabytes in size.

Reply:
It's very unlikely that we'll download something this large. I chose a u64 because that's the type the progress bar we're using to display on the client uses. I could cast this to a u64 on the other end but the progress bar is already not super efficient.

For example: I think right now we download in 8k chunks. That means you typically get a progress message for every 8k chunk since I update on bytes written.